v2/tools/generator/internal/astbuilder/composite_literals.go

```diff
@@ -13,26 +13,35 @@ import (
 type CompositeLiteralBuilder struct {
 	structType dst.Expr
 	elts       []dst.Expr
+	newLines   bool
 }
 
 // NewCompositeLiteralBuilder creates a new instance for initialization of the specified struct
 // structType is an expression to handle both structs from the current package and imported ones requiring qualification
 func NewCompositeLiteralBuilder(structType dst.Expr) *CompositeLiteralBuilder {
 	return &CompositeLiteralBuilder{
 		structType: structType,
+		newLines:   true,
 	}
 }
 
+// WithoutNewLines returns the CompositeLiteralBuilder without NewLines enabled
+func (b *CompositeLiteralBuilder) WithoutNewLines() *CompositeLiteralBuilder {
+	b.newLines = false
+	return b
+}
+
```

Comment:
Do we need this to be explicit, or could we make this automatic?
(I'm not a fan of pushing more cognitive load onto the API unless we have to.)

In `Build()` we could check the length of `elts` and force newlines on the members only if there's more than one.
We'd need to also change the field from `elts []dst.Expr` to `elts []*dst.KeyValueExpr`, but I think that's safe.

We already have some smarts like this elsewhere in `astbuilder`, when generating function calls IIRC.

Reply:
I took a look into doing this. It's totally doable, but the problem is that I am not sure we universally want all CompositeLiterals to be on a single line if they only have one property.

In my mind, it's more about the context they're being used in. If they're being assigned to a variable directly and aren't already deeply nested, then this is ok (or preferable, even):
```
       account.Owner = &genruntime.KnownResourceReference{
               Name: owner.Name,
       }
```

whereas, if they're deeply nested, then I think we prefer this:
```
Watches: []registration.Watch{
			{
				Src:              &source.Kind{Type: &v1.ConfigMap{}},
				MakeEventHandler: watchConfigMapsFactory([]string{".spec.principalIdConfigRef"}, &authorization_v20200801ps.RoleAssignmentList{}),
			},
		},
```

vs

```
Watches: []registration.Watch{
			{
				Src: &source.Kind{
					Type: &v1.ConfigMap{}
				},
				MakeEventHandler: watchConfigMapsFactory([]string{".spec.principalIdConfigRef"}, &authorization_v20200801ps.RoleAssignmentList{}),
			},
		},
```

Since context matters it feels like we should allow the user to choose. Thoughts?

Reply:
Hmm... thinking further on this, I don't hate 
```
       account.Owner = &genruntime.KnownResourceReference{Name: owner.Name}
```

so if folks are ok with that everywhere I can do this (single-line if 1 element) and we can expand later when we run into a case where we really dislike it?

Reply:
Check the "Increase CompositeLiteralBuilder smarts" commit to see the effect of this change. Let me know if you dislike it, otherwise I will keep it.

Reply:
I'm happy with that change - just made some comment requesting a comment. #recursion

Reply:
Resolving this. Verdict is to take the change that @theunrepentantgeek suggested.